modules/1_prepare/outputs.tf

```diff
@@ -42,17 +42,17 @@ output "public_cidr" {
 
 output "bastion_vip" {
   depends_on = [null_resource.bastion_init]
-  value      = local.bastion_count > 1 ? ibm_pi_network_port.bastion_vip[0].pi_network_port_ipaddress : ""
+  value      = local.bastion_count > 1 ? ibm_pi_network_interface.bastion_vip[0].ip_address : ""
 }
 
 output "bastion_internal_vip" {
   depends_on = [null_resource.bastion_init]
-  value      = local.bastion_count > 1 ? ibm_pi_network_port.bastion_internal_vip[0].pi_network_port_ipaddress : ""
+  value      = local.bastion_count > 1 ? ibm_pi_network_interface.bastion_internal_vip[0].ip_address : ""
 }
 
 output "bastion_external_vip" {
   depends_on = [null_resource.bastion_init]
-  value      = local.bastion_count > 1 ? ibm_pi_network_port.bastion_internal_vip[0].public_ip : ""
+  value      = local.bastion_count > 1 ? ibm_pi_network_interface.bastion_internal_vip[0].ip_address : ""
```

Comment:
This will be a problem in ssh command output, DNS host entries, wildcard DNS. Somehow can we get the public IP from network?

Reply:
Hey @yussufsh 

I looked at the [terraform-provider-ibm](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/ibm_pi_network_interface), it's unclear if this is the public IP address. Further looking into the [api](https://cloud.ibm.com/apidocs/power-cloud#v1-networks-network-interfaces-post), it's not clear the API network, instance et cetra is suitable for returning the public ip (I anticipate they built it for VPC entry into the PowerVS workspace).

I'll have to experiment it a bit. I'll do that on Monday.

Thanks,

Paul